Source/Rhetos.Utilities/ApplicationConfiguration/ConfigurationProvider.cs

```diff
@@ -137,8 +139,11 @@ private bool TryGetConfigurationValue(string configurationKey, out object result
 
             if (_configurationValues.TryGetValue(configurationKey, out var entry))
             {
-                if (convertRelativePath && !string.IsNullOrEmpty(entry.BaseFolder) && !string.IsNullOrEmpty(entry.Value as string))
-                    result = Path.Combine(entry.BaseFolder, (string)entry.Value);
+                var baseFolder = Path.GetDirectoryName((entry as FileSourceConfigurationValue)?.FilePath)
+                                 ?? AppDomain.CurrentDomain.BaseDirectory;
+
+                if (convertRelativePath && !string.IsNullOrEmpty(baseFolder) && (entry.Value is string))

```

Comment:
It was `!string.IsNullOrEmpty` and that caused a bug with the way I was testing it. I think this behavior is slightly better because it is in compliance with how `Path.Combine` works (and I think some shells?). Implementing a difference between `"."` and `""` opens up more opportunities for bugs.

Reply:
It seems unintuitive to have a different behavior between null and empty configuration value, but I agree that developers might expect "" to be interpreted as base folder. We can keep to current version.